### PR TITLE
feat: move to npm fontawesome dependency and support fa pro

### PR DESCRIPTION
**Pull request by oplik0**
Community forum reference: https://community.nodebb.org/topic/12809/add-icons-from-fontawesome-to-icons-list-on-nodebb/22?_=1689811394923

This PR moves FontAwesome from a vendored package to an npm dependency, allowing for far simpler updates than before.

Additionally, there are two new configuration options added:
- `fontawesome:pro` is a boolean option for including the `@fortawesome/fontawesome-pro` package instead of `@fortawesome/fontawesome-free` (note that additional setup is required to install it, and as such it'll have to be done manually before enabling this option: https://fontawesome.com/docs/web/setup/packages#_1-configure-access)
- `fontawesome:styles` defines which styles to include, array of style names or `"*"` for all supported styles ~~defaults to `["solid", "brands", "regular"]` (current settings and the styles that are available for free). The reason for adding this setting is to avoid creating unnecessarily big style files in pro installation which would otherwise include 13 times as many icons as soon as it was enabled. I assume this is the reason why FA documentation shows including only the files you need, and as such want to allow the user to do the same.~~
  ~~note that since finding out how to enable pro will require looking at documentation (I'll add these to the `config.json` docs when this PR is ready) or issue tracker anyway, I doubt many people will enable pro and not notice this option :)~~
  Defaults to all supported styles - the mentioned ones for FA free, and all currently supported pro styles if that option is set (addition of light, thin, sharp and duotone styles).
  I didn't think about how browsers handle loading fonts. Since they lazy-load fonts on use, including the small bits of CSS defining font styles for pro families is not actually a big issue, it might just bloat the search results a bit, which is why I'm leaving the setting for those who might want to get rid of some supported styles (e.g. to enforce consistency by only allowing a single FA style)

Due to these changes I have refactored essentially everything fa related on the SCSS front - the vendor folder is removed (though webfonts are still served under their old URL - should I change that to something else? E.g. just `/assets/fontawesome/webfonts`?), and now the loading script and shims are in separate `fontawesome` folder in `public/scss`. Alongside them are separate files for all the styles - the reason for adding these instead of just relying on the config is to work with styles that require more than one import to work (e.g. duotone) and give a shorthand for including all `sharp` styles.

The icon selector now includes a bit of a hack for the new fa styles - it sets the value to multiple classes, just adding the style classes (e.g. `fa-brands`) to the value (so e.g. the microblog icon becomes `fa-microblog fa-brands`).
For systems that want to handle this better, the callback now includes the styles as the third argument - however, from what I've seen NodeBB is using the value of the element instead.

It also now supports FA pro - if the proper config is set, it'll properly search for pro icons and return all of their styles separately. This includes the new `sharp` styles which are a combination style (that is, you need both `fa-solid` and `fa-sharp` classes for a sharp solid icon, because things were too simple as they were).

Also, now the version searched corresponds to the version installed, so there is no need to manually update the search string along with FA package. The only thing to do when updating will be adding a file for any new style they might add (which again, is only done that way because it's not unlikely it might require special handling... Especially since the next one should be sharp-duotone and duotone is what made me do it this way...)

Tasks for this PR:
- [x] move to FA npm package
- [x] add FA pro support
- [x] fix icon selector not adding the style correctly
- [x] add FA pro support to icon filter

### CODE REVIEW

**Comment by nodebb-misty** [comment]
## 💡 Friendly Note

This pull request was made against the `develop` branch, which is reserved for commits destined for a **minor** or **major** release.If your commits simply fixes a bug, please rebase this PR against the `master` instead.

* **patch** releases — bug fixes only
* **minor** releases — new features, enhancements, and bug fixes
* **major** releases — breaking changes, including all of the above

Thanks!

**Comment by barisusakli** [comment]
Are there any breaking changes here?

https://github.com/NodeBB/NodeBB/blob/master/public/src/modules/helpers.common.js#L91 uses `<i class="fa fa-fw ${category.icon}"></i>` and possibly other places hard coded `fa fa-fw` in the template as well. Will this work if someone selects an icon from the other `fa-*` ?

**Comment by oplik0** [comment]
Sorry, I missed this somehow in my notifications.

No - the `fa` prefix was deprecated in FA5, but is kept for compatibility reasons with shims loaded for specific icons in case of NodeBB.

By default it sets the font and weight to the ones from the solid styles, however it will always be overriden by shims (icons from FA4 which don't work with that font weight, changed name or generally require handling different in any way to that default) or... any furhter fa style added, as the `fa` is the very first class described in the CSS (even before styles applied to all fa prefixes). So `fa fa-brands fa-${icon}` ends up being the same thing as `fa-brands fa-${icon}` which is the correct FA5/6 syntax.

One slight issue, that is already present in current selector, is that you can't select a few FA4 icons *directly* - primarily the `-o` suffix outlined icons require selecting from regular style. They have shims, so added programatically they work just fine, it's just that the selector uses FA6 API, so it can't find them since they're now the same icon of different family. So that's actually a fix here :)

File: public/src/modules/iconSelect.js [hunk 1]
```diff
@@ -250,26 +252,40 @@ define('iconSelect', ['benchpress', 'bootbox'], function (Benchpress, bootbox) {
...
-							const iconClass = $('.bootbox .selected').attr('class') || `fa fa-${$('.bootbox #fa-filter').val()}`;
-							const newIconClass = cleanFAClass(iconClass);
-
-							if (newIconClass) {
-								el.removeClass(selected).addClass(newIconClass);
-								el.val(newIconClass);
-								el.attr('value', newIconClass);
+							const iconClass = $('.bootbox .selected')[0]?.classList || [`fa-${$('.bootbox #fa-filter').val()}`];
+							const newIcon = cleanFAClass(iconClass);
+							console.log(`Selected ${JSON.stringify(selected)} new ${JSON.stringify(newIcon)}`);
```
Comment (barisusakli) [inline public/src/modules/iconSelect.js:271]: @oplik0 you can remove this console.log

**Comment by barisusakli** [comment]
thanks @oplik0 💯

**Pull request merged**